Automattic/wp-calypso
# Create the domain upsell steps component

#### Proposed Changes

* This PR adds a new steps navigation component for our domain upsell flow.

Step 1 example:
![step1](https://user-images.githubusercontent.com/140841/217112410-c468d270-3a3e-4b79-9b87-b246513ae18c.png)

Step 2 example:
![step2](https://user-images.githubusercontent.com/140841/217112444-e68d4b33-3bb3-487a-af1b-1c9c12c7c262.png)


#### Testing Instructions



This PR is based on [Prepare the domain page to work with the experiment - backend strategy #73017 ](https://github.com/Automattic/wp-calypso/pull/73017)

* To test this component follow the testing instruction in [Prepare the domain page to work with the experiment - backend strategy #73017 ](https://github.com/Automattic/wp-calypso/pull/73017)
* You should see the navigation header at the top of the Search page. (As seen in the pictures above.)
* To test step 2, modify the component in code to `<DomainAndPlanPackageNavigation step={ 2 } />` and reload the search page. (eventually we'll add this component to the plans page as "step 2").
* Test in different browsers and on mobile.
* Test the "Back" button goes back.

#### Pre-merge Checklist





Related to #
